doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1676,68 +1615,100 @@ Controlling the proof flow
 
-If the goal is :g:`G` and :g:`t` is a subterm of type :g:`T` in the goal,
-then :n:`generalize t` replaces the goal by :g:`forall (x:T), G′` where :g:`G′`
-is obtained from :g:`G` by replacing all occurrences of :g:`t` by :g:`x`. The
-name of the variable (here :g:`n`) is chosen based on :g:`T`.
+   .. example::
 
-.. tacv:: generalize {+ @term}
+      .. coqtop:: reset none
 
-   This is equivalent to :n:`generalize @term; ... ; generalize @term`.
-   Note that the sequence of term :sub:`i` 's are processed from n to 1.
+         Goal forall x y:nat, 0 <= x + y + y.
+         Proof. intros *.
 
-.. tacv:: generalize @term at {+ @natural}
+      .. coqtop:: all abort
 
-   This is equivalent to :n:`generalize @term` but it generalizes only over the
-   specified occurrences of :n:`@term` (counting from left to right on the
-   expression printed using the :flag:`Printing All` flag).
+         Show.
+         generalize (x + y + y).
 
-.. tacv:: generalize @term as @ident
+   .. tacn:: generalize dependent @one_term
 
-   This is equivalent to :n:`generalize @term` but it uses :n:`@ident` to name
-   the generalized hypothesis.
+      Generalizes :n:`@one_term` and all hypotheses that depend on :n:`@one_term`. It
+      clears the generalized hypotheses.
 
-.. tacv:: generalize {+, @term at {+ @natural} as @ident}
+   .. tacn:: dependent generalize_eqs @ident
+      :undocumented:
 
-   This is the most general form of :n:`generalize` that combines the previous
-   behaviors.
+   .. tacn:: dependent generalize_eqs_vars @ident
+      :undocumented:
 
-.. tacv:: generalize dependent @term
+   .. tacn:: generalize_eqs @ident
+      :undocumented:
 
-   This generalizes term but also *all* hypotheses that depend on :n:`@term`. It
-   clears the generalized hypotheses.
+   .. tacn:: generalize_eqs_vars @ident
+      :undocumented:
 
-.. tacn:: evar (@ident : @term)
-   :name: evar
+.. tacn:: evar ( @ident : @term )
+          evar @one_term
+   :name: evar; _
 
-   The :n:`evar` tactic creates a new local definition named :n:`@ident` with type
+   The :n:`evar` tactic creates a new :term:`local definition <context-local definition>`
+   named :n:`@ident` with type
    :n:`@term` in the context. The body of this binding is a fresh existential
    variable.
 
-.. tacn:: instantiate (@ident := @term )
-   :name: instantiate
+.. tacn:: instantiate {? ( @ident := @term ) }
+          instantiate ( @integer := @term ) {? @hloc }
+   :name: instantiate; _
+
+   .. insertprodn hloc hloc
+
+   .. prodn::
+      hloc ::= in %|- *
+      | in @ident
+      | in ( type of @ident )
+      | in ( value of @ident )
+
+   The no argument form is equivalent to :tacn:`idtac`.
+   This form (only) is deprecated in 8.16.
 
-   The instantiate tactic refines (see :tacn:`refine`) an existential variable
+   .. deprecated:: 8.16
+
+   .. todo PR perhaps we can change the grammar to use "natural"?
```

Comment:
Yes, that sounds like a good idea, unless there is some undocumented meaning to using a negative number here (like counting from left to right, maybe?).

Reply:
Negative values and zero values are not permitted

  if n <= 0 then user_err Pp.(str "Incorrect existential variable index.");

Reply:
See #16642